### PR TITLE
allow keep-alive to be lowercase

### PR DESCRIPTION
**Pull request by akovari**

### CODE REVIEW

File: src/elli_http.erl [hunk 1]
```diff
@@ -558,6 +558,7 @@ connection_token(#req{version = {1, 1}, headers = Headers}) ->
 connection_token(#req{version = {1, 0}, headers = Headers}) ->
     case proplists:get_value(<<"Connection">>, Headers) of
         <<"Keep-Alive">> -> <<"Keep-Alive">>;
+        <<"keep-alive">> -> <<"Keep-Alive">>;
```
Comment (yurrriq) [inline src/elli_http.erl]: Should we make these two lowercase too? And the match on 572?

Comment (yurrriq) [reply src/elli_http.erl]: If we do that, it looks like we'd have to update several tests.

**Comment by yurrriq** [comment]
Thoughts on https://github.com/elli-lib/elli/pull/9/commits/f82fd2b6df8d7a892387ddaf780c416fae3e4e70 and https://github.com/elli-lib/elli/pull/9/commits/810eab7a45654ec756c14d01b1d76d8def7de3a1, @elli-lib/team?

**Comment by deadtrickster** [comment]
I'm sorry but this is a dead end - headers are cases insensitive by the RFC and have to be normalized right after parsing

**Comment by deadtrickster** [comment]
Actually this is usually done during parsing

**Comment by yurrriq** [comment]
I'm not sure I understand the need for this patch and it sounds like there isn't one. Is there some specific problem this solves for you, @akovari?

**Comment by akovari** [comment]
Well, I know ideally this should be handled in a case-insensitive way but I didn't want to do such major change as I am not sure how much I could potentially break. Maybe some time in the future.
Current expectation of Keep-Alive breaks in our web application, which uses the fetch() API in Javascript, which at least in Chrome, seems to generate lower-case headers. So yes, this helps a lot.

**Comment by yurrriq** [comment]
It's very late here and I'm off to bed, but perhaps a better approach then would be to normalize all the headers higher up the call stack. Normalizing to lower case seems sound to me.

**Comment by akovari** [comment]
Maybe we could also use this: https://github.com/ninenines/cowlib/blob/master/src/cow_http_hd.erl

**Comment by yurrriq** [comment]
Closing this stale PR. Feel free to re-open as desired.

**Pull request closed without merging**